run_project_tests.py

```diff
@@ -360,83 +355,60 @@ def run_test_inprocess(testdir):
         os.chdir(old_cwd)
     return max(returncode_test, returncode_benchmark), mystdout.getvalue(), mystderr.getvalue(), test_log
 
-def parse_test_args(testdir):
-    args = []
-    try:
-        with open(os.path.join(testdir, 'test_args.txt'), 'r') as f:
-            content = f.read()
-            try:
-                args = literal_eval(content)
-            except Exception:
-                raise Exception('Malformed test_args file.')
-            args = stringlistify(args)
-    except FileNotFoundError:
-        pass
-    return args
-
 # Build directory name must be the same so Ccache works over
 # consecutive invocations.
-def create_deterministic_builddir(src_dir, name):
+def create_deterministic_builddir(test: TestDef) -> str:
     import hashlib
-    if name:
-        src_dir += name
+    src_dir = test.path.as_posix()
+    if test.name:
+        src_dir += test.name
     rel_dirname = 'b ' + hashlib.sha256(src_dir.encode(errors='ignore')).hexdigest()[0:10]
     os.mkdir(rel_dirname)
     abs_pathname = os.path.join(os.getcwd(), rel_dirname)
     return abs_pathname
 
-def run_test(skipped, testdir, name, extra_args, compiler, backend, flags, commands, should_fail):
-    if skipped:
+def run_test(test: TestDef, extra_args, compiler, backend, flags, commands, should_fail):
+    if test.skip:
         return None
-    with AutoDeletedDir(create_deterministic_builddir(testdir, name)) as build_dir:
+    with AutoDeletedDir(create_deterministic_builddir(test)) as build_dir:
         with AutoDeletedDir(tempfile.mkdtemp(prefix='i ', dir=os.getcwd())) as install_dir:
             try:
-                return _run_test(testdir, build_dir, install_dir, extra_args, compiler, backend, flags, commands, should_fail)
+                return _run_test(test, build_dir, install_dir, extra_args, compiler, backend, flags, commands, should_fail)
             finally:
                 mlog.shutdown() # Close the log file because otherwise Windows wets itself.
 
-def pass_prefix_to_test(dirname):
-    if '39 prefix absolute' in dirname:
+def pass_prefix_to_test(dirname: Path):
+    if '39 prefix absolute' in dirname.name:
```

Comment:
While you're hear, the return type is `bool`, and this can be simplified to simply 
```python
def pass_prefix_to_test(dirname: Path) -> bool:
    return '39 prefix absolute' not in dirname.name
```

Reply:
Even better, I removed all those functions and replaced them with a `do_not_pass` entry in `test.json`.